app/src/main/java/io/homeassistant/companion/android/util/DisabledLocationHandler.kt

```diff
@@ -0,0 +1,59 @@
+package io.homeassistant.companion.android.util
+
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.location.LocationManager
+import android.os.Build.VERSION
+import android.os.Build.VERSION_CODES
+import android.provider.Settings
+import androidx.appcompat.app.AlertDialog
+import io.homeassistant.companion.android.R
+
+object DisabledLocationHandler {
+    fun isLocationEnabled(context: Context, fineLocation: Boolean): Boolean {
+        val lm: LocationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+
+        return if (VERSION.SDK_INT >= VERSION_CODES.P) {
+            lm.isLocationEnabled
+        } else {
+            (!fineLocation && lm.isProviderEnabled(LocationManager.NETWORK_PROVIDER) || (fineLocation && lm.isProviderEnabled(LocationManager.GPS_PROVIDER)))
+        }
+    }
+
+    fun containsFineLocationPermission(permissions: Array<String>): Boolean {
+        return permissions.contains(Manifest.permission.ACCESS_FINE_LOCATION)
+    }
+
+    fun containsCoarseLocationPermission(permissions: Array<String>): Boolean {
+        return permissions.contains(Manifest.permission.ACCESS_COARSE_LOCATION)
+    }
+
+    fun showLocationDisabledWarnDialog(activity: Activity, context: Context, message: String, withDisableOption: Boolean = false, callback: (() -> Unit)? = null) {
```

Comment:
Rather than having multiple options for the dialog lets just use the `location_disabled_general_message` one and list all sensors that will be disabled?

Reply:
You are right. Doesn't hurt to have this information on the popup.